src/server/ua_services_nodemanagement.c

```diff
@@ -692,50 +688,28 @@ AddNode_addRefs(UA_Server *server, UA_Session *session, const UA_NodeId *nodeId,
          * that type has the same nodeClass from checkParentReference. */
         if(node->nodeClass == UA_NODECLASS_VARIABLE) {
             if(((const UA_VariableTypeNode*)type)->isAbstract) {
-                /* Get subtypes of the parent reference types */
-                UA_NodeId *parentTypeHierachy = NULL;
-                size_t parentTypeHierachySize = 0;
-                retval = getLocalRecursiveHierarchy(server, parentReferences, UA_PARENT_REFERENCES_COUNT,
```

Comment:
Just removing this line will not work (as you can see in the CI output).

With your new version you only check for references of type `subtypeId`. Custom nodesets can define subtypes of this parent reference type. When getting the type hierarchy and calling isNodeInTree we need to make sure that we include the `subtypeId` reference and all its reference subtypes.

Same a few lines below for the Object.